### ISSUE TITLE
System.Numerics.Vectors missing intrinsics

### ISSUE THREAD

**Issue by dangi12012** [dotnet/runtime#25622]
There are 100s of Vector<T> commands that would map to a single AVX or SSE instruction that are not present in System.Numerics.Vector. Also there are no Vertical Operators except Dot and BiggerThanAll.
Here are all:
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#

Would help performance if some Vertical Operators would be supported. Like Max, Min, Sum,
```
public static float Vector<float> Max()
public static float Vector<float> Min()
public static float Vector<float> Sum()

//Max:
movhlps xmm1,xmm0         ; Move top two floats to lower part of xmm1
maxps   xmm0,xmm1         ; Get maximum of the two sets of floats
pshufd  xmm1,xmm0,$55     ; Move second float to lower part of xmm1
maxps   xmm0,xmm1         ; Get minimum of the two remaining floats
```
Vector could also be a special type in C# like Span<T> to support 0 copy operations and casts.

**Comment by GrabYourPitchforks** [comment]
> That being said, there are likely still some "general-purpose" algorithms that we could expose on `System.Numerics.Vector<T>`. This would ideally be reviewed on a case-by-case basis with specific use-cases/scenarios provided.

@dangi12012 - can you provide a list of specific APIs you'd like to see added to the `Vector<T>` class?

**Comment by tannergooding** [comment]
Going to close this as unactionable.

If someone feels strongly about a particular missing API, please feel free to open a new issue. Our API review process is detailed [here](https://github.com/dotnet/runtime/blob/master/docs/project/api-review-process.md) with a "good example" being listed under step 1. Ideally the new issue would provide many of the sections given, but would at a minimum provide the base rationale and proposed API surface sections.

The System.Runtime.Intrinsics feature and the new APIs for converting between the intrinsic vector and System.Numerics.Vector types should unblock any users who immediately need such functionality, as they can implement their own extension methods.
